src/circulating_supply/circulating_supply_provider.rs

```diff
@@ -0,0 +1,150 @@
+        Ok(None) => {
+            info!(
+                target: crate::INDEXER_FOR_EXPLORER,
+                "Computing circulating supply for the timestamp {}", block_timestamp
+            );
+        }
+        Ok(Some(supply)) => {
+            info!(
+                target: crate::INDEXER_FOR_EXPLORER,
+                "Circulating supply for the timestamp {} was already computed: {}",
+                block_timestamp,
+                supply
+            );
+            return;
+        }
+        Err(err) => {
+            panic!("Error {}", err);
+        }
+    }
+
+    let lockups = collect_lockups_for_block(block_height, &pool).await;
+
+    let mut total_locked_amount: u128 = 0;
+    let mut active_locked_accounts: u128 = 0;
+
+    for lockup in lockups.iter() {
+        let state = get_account_state(&view_client, lockup, block_height).await;
+        let is_lockup_with_bug = is_bug_inside(
+            &get_code_version(&view_client, lockup, block_height).await,
+            lockup,
+        );
+        let locked_amount = state
+            .get_locked_amount(block_timestamp, is_lockup_with_bug)
+            .0;
+        total_locked_amount += locked_amount;
+        if locked_amount > 0 {
+            active_locked_accounts += 1;
+        }
+    }
+
+    let foundation_locked_accs = ["lockup.near", "contributors.near"];
+    let mut foundation_locked: u128 = 0;
+    for acc in foundation_locked_accs.iter() {
+        foundation_locked += get_user_balance(view_client.clone(), acc, block_height).await;
+    }
+
+    let circulating_supply: u128 = total_supply - foundation_locked - total_locked_amount;
+
+    let stats = CirculatingSupply {
+        block_timestamp: block.block_timestamp,
+        block_hash: block.block_hash,
+        value: BigDecimal::from_str(circulating_supply.to_string().as_str())
+            .expect("`value` expected to be u128"),
+        total_supply: BigDecimal::from_str(total_supply.to_string().as_str())
+            .expect("`total_supply` expected to be u128"),
+        lockups_number: BigDecimal::from_str(lockups.len().to_string().as_str())
+            .expect("`lockups_number` expected to be u128"),
+        active_lockups_number: BigDecimal::from_str(active_locked_accounts.to_string().as_str())
+            .expect("`active_lockups_number` expected to be u128"),
+        foundation_locked_supply: BigDecimal::from_str(foundation_locked.to_string().as_str())
+            .expect("`foundation_locked_supply` expected to be u128"),
+        lockups_locked_supply: BigDecimal::from_str(total_locked_amount.to_string().as_str())
+            .expect("`lockups_locked_supply` expected to be u128"),
+    };
+
+    add_circulating_supply(&pool, &stats).await;
+    info!(
+        target: crate::INDEXER_FOR_EXPLORER,
+        "Circulating supply for the timestamp {} is {}", block_timestamp, circulating_supply
+    );
+}
+
+async fn find_last_yesterday_block_for(
```

Comment:
Consider replace "yesterday" to be more clear :)

```suggestion
async fn find_last_yesterday_block_for_timestamp(
```

Reply:
Yesterday sounds like the most suitable option, sorry 😄 
This is exactly what we want to find there

Tried to play with `day_before`, it always sounds weirder